### PR TITLE
Add learning rate decay test

### PR DESCRIPTION
**Pull request by chengduoZH**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_parallel_executor_seresnext.py [hunk 1]
```diff
@@ -131,27 +135,55 @@ def SE_ResNeXt50Small(batch_size=2, use_feed=False):
...
+            use_reduce=use_reduce,
+            optimizer=_optimizer)
+
+    def test_resnet_with_learning_rate_decay(self):
+        self.check_resnet_convergence_with_learning_rate_decay(True, False)
+        self.check_resnet_convergence_with_learning_rate_decay(
+            False, False, iter=5)
+
+    def test_resnet_with_new_strategy_with_learning_rate_decay(self):
+        self.check_resnet_convergence_with_learning_rate_decay(True, True)
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_parallel_executor_seresnext.py]: Test convergence may not be precise, if decay operators is not correctly put in the interal ssa graph, the network may still convergence?

Comment (chengduoZH) [reply python/paddle/fluid/tests/unittests/test_parallel_executor_seresnext.py]: > if decay operators is not correctly put in the interal ssa graph

If so, it means that there is a problem with our Graph building, we should fix it in time.

Comment (typhoonzero) [reply python/paddle/fluid/tests/unittests/test_parallel_executor_seresnext.py]: So there will be another test for that?

Comment (chengduoZH) [reply python/paddle/fluid/tests/unittests/test_parallel_executor_seresnext.py]: We can compare the result of ParallelExecutor and Executor, I add this in next commit.

**Comment by typhoonzero** [comment]
The test seems failed?

**Comment by chengduoZH** [comment]
@typhoonzero  This is because "@LR_DECAY_COUNTER@" is shared by multi threads. Now it is fixed.

File: paddle/fluid/framework/parallel_executor.cc [hunk 2]
```diff
@@ -218,7 +218,10 @@ void ParallelExecutor::BCastParamsToDevs(
 
         auto local_scope = member_->local_scopes_[i];
         auto *t = local_scope->Var(var)->GetMutable<LoDTensor>();
-        if (member_->use_all_reduce_ || member_->use_cuda_) {
+
+        // FIXME(zcd): LR_DECAY_COUNTER should not be shared. This is a hot fix.
+        if (member_->use_all_reduce_ || member_->use_cuda_ ||
+            var == "@LR_DECAY_COUNTER@") {
```
Comment (typhoonzero) [inline paddle/fluid/framework/parallel_executor.cc:224]: The name may not be `@ LR_DECAY_COUNTER@` should keep this in mind and fix it later.

**Review by typhoonzero — APPROVED** [review]
